src/FluentContracts/Contracts/BaseContract.cs

```diff
@@ -63,5 +63,29 @@ public Linker<TContract> Satisfy<T, TException>(Func<T, bool> customCondition, s
         Validator.CheckGenericCondition<T, TException>(customCondition, (T)ArgumentValue, message);
         return _linker;
     }
+
+    public Linker<TContract> HasProperty(string propertyName, string? message = null)
+    {
+        Validator.CheckForNotNull(ArgumentValue, ArgumentName, message);
+        Validator.ContainsPropertyByName(ArgumentValue, propertyName, message);
+
+        return _linker;
+    }
+
+    public Linker<TContract> HasPropertyWithValue(string propertyName, object value, string? message = null)
```

Comment:
Please, refrain from using `object` unless completely necessary. Here you can change it to:
```suggestion
    public Linker<TContract> HasPropertyWithValue<TValue>(string propertyName, TValue value, string? message = null)
```